docs/blog/rustport/index.html

```diff
@@ -0,0 +1,354 @@
+
+<p>Most of the work was done by 7 people (going by those with at least 10 commits to “.rs” files), but we got a lot of help from interested community members.</p>
+
+<p>The delay after that was down to a few reasons:</p>
+
+<ol>
+  <li>The “second 90%” - testing that everything worked. We flushed out a lot of bugs in this time, and if we made a release at that time it would have been a bad one.</li>
+  <li>Having something to release that’s visible to users - there’s no point in making a release that does the same thing in new code, you need it to do different things.
+So we held off until we had something.</li>
+  <li>Simple availability - sometimes, some of us took time off.</li>
+</ol>
+
+<p>So if you are trying to draw any conclusions from this, consider the context: A group of people working on a thing in their free time,
+diverting some effort to work on something else, <em>and</em> deciding that after the work is finished it actually isn’t.</p>
+
+<h2 id="the-gripes">The Gripes</h2>
+
+<p>It won’t surprise anyone who has spent any time on this world of ours that Rust is not, in fact, perfect. We have some gripes with it.</p>
+
+<p>Chief among them is how Rust handles portability. While it offers many abstractions over systems, allowing you to target a variety of systems with the same code,
+when it comes to <em>adapting</em> your code to systems at a lower-level, it’s all based on enumerating systems by hand, using checks like <code class="language-plaintext highlighter-rouge">#[cfg(any(target_os = "freebsd", target_os = "netbsd", target_os = "openbsd"))]</code>.</p>
+
+<p>This is an imperfect solution, allowing you to miss systems and ignoring version differences entirely. From what we can tell, if FreeBSD 12 gains a function that we want to use, libc would add it, but calling it would then fail on FreeBSD 11 without a good way to check, at the moment.</p>
+
+<p>But listing targets in our code is also fundamentally duplicating work that the libc crate (in our case) has already done. If you want to call libc::X, which is only defined on systems A, B and C, you need to put in that check for A, B and C yourself and if libc adds system D you need to add it as well. Instead of doing that, we are using our own <a href="https://github.com/mqudsi/rsconf">rsconf</a> crate to do compile-time feature detection in build.rs.</p>
+
+<p>Most of this would be solved if Rust had some form of saying “compile this if that function exists” - <code class="language-plaintext highlighter-rouge">#[cfg(has_fn = "fstatat")]</code>. With that, the libc crate could do whatever checks it wants and fish would just follow what it did, and we could remove a lot of the use for rsconf. It would not really help support older distributions that lack some features, tho. That could be solved by something like the <a href="https://github.com/rust-lang/rfcs/pull/3036">min_target_API_version</a> cfg.</p>
```

Comment:
Sounds like the correct issue here would be https://github.com/rust-lang/rust/issues/64797 (`#[cfg(accessible(::path::to::thing))]`)

Reply:
Ah, that's probably the one I was looking for.

I knew *something* here had been proposed but I forgot what it was exactly.